panic in docker module: interface conversion int64 uint64

Ubuntu 16.04, telegraf: 1.1.0, installed from ofiicial repo.

If in docker plugin `total = true`, telegraf fails with mesaage:

```
ноя 15 14:05:21 newhope telegraf[10797]: panic: interface conversion: interface is int64, not uint64
ноя 15 14:05:21 newhope telegraf[10797]: goroutine 109 [running]:
ноя 15 14:05:21 newhope telegraf[10797]: panic(0xf24700, 0xc420050940)
ноя 15 14:05:21 newhope telegraf[10797]:         /usr/local/go/src/runtime/panic.go:500 +0x1a1
ноя 15 14:05:21 newhope telegraf[10797]: github.com/influxdata/telegraf/plugins/inputs/docker.gatherBlockIOMetrics(0xc4201d4000, 0x18dfd20, 0xc4202d9650, 0xc420288390, 0xecfbce2f1, 0xc419c899a3, 0x18acdc0, 0xc42010d800, 0x40, 0xd472062c93e10101)
ноя 15 14:05:21 newhope telegraf[10797]:         /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:492 +0x2109
ноя 15 14:05:21 newhope telegraf[10797]: github.com/influxdata/telegraf/plugins/inputs/docker.gatherContainerStats(0xc4201d4000, 0x18dfd20, 0xc4202d9650, 0xc420288390, 0xc42010d800, 0x40, 0x18d0101)
ноя 15 14:05:21 newhope telegraf[10797]:         /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:385 +0x30b1
ноя 15 14:05:21 newhope telegraf[10797]: github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).gatherContainer(0xc420196360, 0xc42010d800, 0x40, 0xc42010d840, 0x1, 0x4, 0xc4205b2d80, 0xc, 0xc4200f9090, 0x47, ...)
ноя 15 14:05:21 newhope telegraf[10797]:         /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:264 +0x68f
ноя 15 14:05:21 newhope telegraf[10797]: github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather.func1(0xc4205d2450, 0xc420196360, 0x18dfd20, 0xc4202d9650, 0xc42010d800, 0x40, 0xc42010d840, 0x1, 0x4, 0xc4205b2d80, ...)
ноя 15 14:05:21 newhope telegraf[10797]:         /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:127 +0xc5
ноя 15 14:05:21 newhope telegraf[10797]: created by github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather
ноя 15 14:05:21 newhope telegraf[10797]:         /home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:132 +0x2dd
ноя 15 14:05:21 newhope systemd[1]: telegraf.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
ноя 15 14:05:21 newhope systemd[1]: telegraf.service: Unit entered failed state.
ноя 15 14:05:21 newhope systemd[1]: telegraf.service: Failed with result 'exit-code'.
ноя 15 14:05:21 newhope systemd[1]: telegraf.service: Service hold-off time over, scheduling restart.
ноя 15 14:05:21 newhope systemd[1]: Stopped The plugin-driven server agent for reporting metrics into InfluxDB.
ноя 15 14:05:21 newhope systemd[1]: Started The plugin-driven server agent for reporting metrics into InfluxDB.
```


Tuning `total = false` fix the problem.

I am experiencing the same crash - telegraf running in a ubuntu 14.4 docker container, mounting /var/run into a '--privileged' docker container - full trace:
`2016/12/07 20:52:50 D! Input [inputs.cpu] gathered metrics, (10s interval) in 567.741µs
panic: interface conversion: interface is int64, not uint64

goroutine 42 [running]:
panic(0xf24700, 0xc420052980)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/influxdata/telegraf/plugins/inputs/docker.gatherBlockIOMetrics(0xc4200f0000, 0x18dfd20, 0xc42000ccc0, 0xc42024da70, 0xecfda6da4, 0xc407be3aee, 0x18acdc0, 0xc42006f0c0, 0x40, 0x9289f8f245e60101)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:492 +0x2109
github.com/influxdata/telegraf/plugins/inputs/docker.gatherContainerStats(0xc4200f0000, 0x18dfd20, 0xc42000ccc0, 0xc42024da70, 0xc42006f0c0, 0x40, 0x18d0101)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:385 +0x30b1
github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).gatherContainer(0xc42006c2a0, 0xc42006f0c0, 0x40, 0xc42006f100, 0x1, 0x4, 0xc4204505a0, 0x13, 0xc420013ae0, 0x47, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:264 +0x68f
github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather.func1(0xc4206ac040, 0xc42006c2a0, 0x18dfd20, 0xc42000ccc0, 0xc42006f0c0, 0x40, 0xc42006f100, 0x1, 0x4, 0xc4204505a0, ...)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:127 +0xc5
created by github.com/influxdata/telegraf/plugins/inputs/docker.(*Docker).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/docker/docker.go:132 +0x2dd`

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -64,6 +64,7 @@ plugins, not just statsd.
 - [#1825](https://github.com/influxdata/telegraf/issues/1825): Consul plugin: add check_id as a tag in metrics to avoid overwrites.
 - [#1973](https://github.com/influxdata/telegraf/issues/1973): Partial fix: logparser CLF pattern with IPv6 addresses.
 - [#1975](https://github.com/influxdata/telegraf/issues/1975) & [#2102](https://github.com/influxdata/telegraf/issues/2102): Fix thread-safety when using multiple instances of the statsd input plugin.
+- [#2027](https://github.com/influxdata/telegraf/issues/2027): docker input: interface conversion panic fix.
 
 ## v1.1.2 [2016-12-12]
 

--- a/plugins/inputs/docker/docker.go
+++ b/plugins/inputs/docker/docker.go
@@ -368,11 +368,22 @@ func gatherContainerStats(
 				if field == "container_id" {
 					continue
 				}
+
+				var uintV uint64
+				switch v := value.(type) {
+				case uint64:
+					uintV = v
+				case int64:
+					uintV = uint64(v)
+				default:
+					continue
+				}
+
 				_, ok := totalNetworkStatMap[field]
 				if ok {
-					totalNetworkStatMap[field] = totalNetworkStatMap[field].(uint64) + value.(uint64)
+					totalNetworkStatMap[field] = totalNetworkStatMap[field].(uint64) + uintV
 				} else {
-					totalNetworkStatMap[field] = value
+					totalNetworkStatMap[field] = uintV
 				}
 			}
 		}
@@ -491,11 +502,22 @@ func gatherBlockIOMetrics(
 				if field == "container_id" {
 					continue
 				}
+
+				var uintV uint64
+				switch v := value.(type) {
+				case uint64:
+					uintV = v
+				case int64:
+					uintV = uint64(v)
+				default:
+					continue
+				}
+
 				_, ok := totalStatMap[field]
 				if ok {
-					totalStatMap[field] = totalStatMap[field].(uint64) + value.(uint64)
+					totalStatMap[field] = totalStatMap[field].(uint64) + uintV
 				} else {
-					totalStatMap[field] = value
+					totalStatMap[field] = uintV
 				}
 			}
 		}

